File: app/src/main/java/eu/warble/pjapp/ui/list/ListActivity.java
package eu.warble.pjapp.ui.list;


import android.os.Bundle;
import android.support.v7.app.ActionBar;
import android.view.MenuItem;
import android.widget.ListAdapter;
import android.widget.ListView;

import eu.warble.pjapp.R;
import eu.warble.pjapp.ui.base.BaseActivity;

public class ListActivity extends BaseActivity<ListPresenter> {

    private ListView listView;

    @Override
    protected ListPresenter createPresenter() {
        return new ListPresenter(this);
    }

    @Override
    protected void initView(Bundle savedInstanceState) {
        setTheme(R.style.AppTheme);
        setContentView(R.layout.activity_list);
        initActionBar();
        listView = findViewById(R.id.listView);
    }

    public void setAdapter(ListAdapter adapter){
        listView.setAdapter(adapter);
    }

    public void setTitle(String title){
        ActionBar bar = getSupportActionBar();
        if (bar != null){
            bar.setTitle(title);
        }
    }

    private void initActionBar(){
        ActionBar bar = getSupportActionBar();
        if (bar != null){
            bar.setHomeButtonEnabled(true);
            bar.setDisplayHomeAsUpEnabled(true);
        }
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()){
            case android.R.id.home:
                this.finish();
                return true;
            default:
                return false;
        }
    }
}

File: app/src/main/java/eu/warble/pjappkotlin/mvp/list/ListActivity.kt
package eu.warble.pjappkotlin.mvp.list

import android.os.Bundle
import android.support.v7.widget.DividerItemDecoration
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.view.MenuItem
import eu.warble.pjappkotlin.R
import eu.warble.pjappkotlin.mvp.ApplicationNavigator
import eu.warble.pjappkotlin.mvp.BaseActivity
import eu.warble.pjappkotlin.utils.Injection
import kotlinx.android.synthetic.main.activity_list.recyclerView


class ListActivity : BaseActivity(), ListContract.View {

    override lateinit var presenter: ListContract.Presenter
    val applicationNavigator by lazy {
        ApplicationNavigator(this)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setTheme(R.style.AppTheme)
        setContentView(R.layout.activity_list)

        supportActionBar?.apply {
            setHomeButtonEnabled(true)
            setDisplayHomeAsUpEnabled(true)
        }

        presenter = ListPresenter(
                Injection.provideStudentDataRepository(applicationContext),
                this,
                applicationContext,
                intent.getStringExtra("adapterType")
        )

        recyclerView.addItemDecoration(DividerItemDecoration(recyclerView.context, DividerItemDecoration.VERTICAL))
    }

    override fun onResume() {
        super.onResume()
        presenter.start()
    }

    override fun setAdapter(adapter: RecyclerView.Adapter<*>) {
        recyclerView.layoutManager = LinearLayoutManager(this)
        recyclerView.adapter = adapter
        recyclerView.invalidate()
    }

    override fun setTitle(title: String) {
        supportActionBar?.title = title
    }

    override fun onOptionsItemSelected(item: MenuItem?): Boolean {
        if (item?.itemId == android.R.id.home) {
            applicationNavigator.goBack()
            return true
        }
        return super.onOptionsItemSelected(item)
    }
}